plugins/modules/win_timezone.ps1

```diff
@@ -3,71 +3,75 @@
 # Copyright: (c) 2015, Phil Schwartz <schwartzmx@gmail.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#AnsibleRequires -CSharpUtil Ansible.Basic
 
-$params = Parse-Args $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
-$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+$spec = @{
+    options = @{
+        timezone = @{
+            required = $true
+            type = "str"
+        }
+    }
+    supports_check_mode = $true
+}
 
-$timezone = Get-AnsibleParam -obj $params -name "timezone" -type "str" -failifempty $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
-$result = @{
-    changed = $false
-    previous_timezone = $timezone
-    timezone = $timezone
-}
+$timezone = $module.Params.timezone
 
-Try {
-    # Get the current timezone set
-    $result.previous_timezone = $(tzutil.exe /g)
-    If ($LASTEXITCODE -ne 0) {
-        Throw "An error occurred when getting the current machine's timezone setting."
-    }
+$module.Result.previous_timezone = $null
+$module.Result.timezone = $timezone
```

Comment:
The module result is located under `$module.Result`. We assign the return values accordingly.